src/file-selector.ts

```diff
@@ -147,64 +147,63 @@ function fromDataTransferItem(
   return Promise.resolve(fwp);
 }
 
-// https://developer.mozilla.org/en-US/docs/Web/API/FileSystemEntry
-async function fromEntry(entry: any) {
-  return entry.isDirectory ? fromDirEntry(entry) : fromFileEntry(entry);
+async function fromEntry(
+  entry: FileSystemEntry,
+): Promise<FileWithPath | FileArray[]> {
+  if (isFileSystemDirectoryEntry(entry)) {
+    return fromDirectoryEntry(entry);
+  } else if (isFileSystemFileEntry(entry)) {
+    return fromFileEntry(entry);
+  }
+
+  return [];
 }
 
-// https://developer.mozilla.org/en-US/docs/Web/API/FileSystemDirectoryEntry
-function fromDirEntry(entry: any) {
+async function fromDirectoryEntry(
```

Comment:
I can rename the function back if you like, I only changed it so that 'directory' is fully written out as that matches what the APIs we call here do as well. I've removed the comments because the type definitions already have all the same links:

![A screenshot of the type definitions for FileSystemDirectoryEntry with links to the MDN documentation in the description](https://github.com/user-attachments/assets/6474431c-6791-4b34-ae44-2e67820c4867)

Let me know what you want to do with this.

Reply:
Function name is ok (can stay renamed). But keep in mind that not everyone may be using an IDE which can point them to the type definitions.

Reply:
> But keep in mind that not everyone may be using an IDE which can point them to the type definitions.

Fair, but this is supported by the language server protocol, which any decent IDE that does more than syntax highlighting supports.

I'll consider this one resolved :+1: